spark/src/test/java/org/apache/iceberg/spark/source/TestSparkTableUtil.java

```diff
@@ -131,17 +133,17 @@ public void after() throws IOException {
   }
 
   @Test
-  public void testPartitionScan() {
-    Seq<SparkPartition> partitions = SparkTableUtil.getPartitions(spark, qualifiedTableName);
+  public void testPartitionScan() throws NoSuchDatabaseException, NoSuchTableException, ParseException {
```

Comment:
We avoid throwing checked exceptions. Although this was the old behavior, I'm reluctant to add them to the signature of the SparkTableUtil methods because they weren't there previously. I think it would be better to wrap these using equivalent Iceberg exception classes that are unchecked, like NoSuchNamespaceException, NoSuchTableException, and ValidationException.

Reply:
@aokolnychyi, thoughts?

Reply:
+1 to keeping it consistent with Iceberg conventions regarding use of UncheckedExceptions.

Reply:
I am also +1. We have multiple examples in the codebase. For example, `HiveTableOperations` wraps a checked `AlreadyExistsException` from Hive into our own unchecked `AlreadyExistsException`.

Reply:
Also, because we may need the same wrapping logic in many places, it probably makes sense to define a function for that and reuse in all places.

Reply:
That was the first of the two dilemmas I had. The second being the return types as java.util collections. Let me change all of them to unchecked exceptions.

Reply:
I have addressed this here: https://github.com/apache/iceberg/pull/1126/commits/556754c65bf1d9cd97bb8e51263f8c7db8a87b95#diff-fa1251337fb4092082819225c9e40a80R39

@aokolnychyi - Though I have created a class ExceptionUtil.java which does the mapping of specific *checked exceptions* to *unchecked equivalents*, I have to make it in the spark module because the exceptions depend on Spark libraries. Let me know if you want to move it to elsewhere.